build/source/engine/groundwatr.f90

```diff
@@ -419,117 +328,81 @@ subroutine computeBaseflow(&
 
  ! compute the smoothing function (-)
- if(availStorage < xMinEval)then
-  ! (compute the logistic function)
+ if (availStorage < xMinEval) then
+  ! compute the logistic function
   expF = exp((availStorage - xCenter)/xWidth)
   logF = 1._rkind / (1._rkind + expF)
   ! (compute the derivative in the logistic function w.r.t. volumetric liquid water content in each soil layer)
-  dLogFunc_dLiq(1:nSoil) = mLayerDepth(1:nSoil)*(expF/xWidth)/(1._rkind + expF)**2._rkind
+  dLogFunc_dLiq(1:nSoil) = mLayerDepth(1:nSoil)*(expF/xWidth)/(1._rkind + expF)**2_i4b !!integer exponent
  else
   logF             = 0._rkind
   dLogFunc_dLiq(:) = 0._rkind
  end if
 
- ! compute the exfiltartion (m s-1)
+ ! compute the exfiltration (m s-1)
  if(totalColumnInflow > totalColumnOutflow .and. logF > tiny(1._rkind))then
   scalarExfiltration = logF*(totalColumnInflow - totalColumnOutflow)  ! m s-1
-  !write(*,'(a,1x,10(f30.20,1x))') 'scalarExfiltration = ', scalarExfiltration
  else
   scalarExfiltration = 0._rkind
  end if
 
- ! check
- !write(*,'(a,1x,10(f30.20,1x))') 'zActive(1), soilDepth, availStorage, logF, scalarExfiltration = ', &
- !                                 zActive(1), soilDepth, availStorage, logF, scalarExfiltration
- !if(scalarExfiltration > tiny(1.0_rkind))  pause 'exfiltrating'
-
- ! compute the baseflow in each layer (m s-1)
- mLayerBaseflow(1:nSoil) = (mLayerColumnOutflow(1:nSoil) - mLayerColumnInflow(1:nSoil))/HRUarea
+ mLayerBaseflow(1:nSoil) = (mLayerColumnOutflow(1:nSoil) - mLayerColumnInflow(1:nSoil))/HRUarea ! compute the baseflow in each layer (m s-1)
 
- ! compute the total baseflow
- qbTotal = sum(mLayerBaseflow)
+ qbTotal = sum(mLayerBaseflow) ! compute the total baseflow
 
  ! add exfiltration to the baseflow flux at the top layer
  mLayerBaseflow(1)      = mLayerBaseflow(1) + scalarExfiltration
  mLayerColumnOutflow(1) = mLayerColumnOutflow(1) + scalarExfiltration*HRUarea
 
- ! test
- if(printFlag)then
-  xDepth = sum(mLayerDepth(ixSaturation:nSoil)*(mLayerVolFracLiq(ixSaturation:nSoil) - fieldCapacity))/sum(theta_sat(ixSaturation:nSoil) - fieldCapacity)  ! "effective" water table thickness (m)
-  xTran  = tran0*(xDepth/soilDepth)**zScale_TOPMODEL  ! transmissivity for the entire aquifer (m2 s-1)
-  xFlow  = xTran*tan_slope*contourLength/HRUarea   ! total column outflow (m s-1)
-  print*, 'ixSaturation = ', ixSaturation
-  write(*,'(a,1x,5(f30.20,1x))') 'tran0, soilDepth                 = ', tran0, soilDepth
-  write(*,'(a,1x,5(f30.20,1x))') 'surfaceHydCond, zScale_TOPMODEL  = ', surfaceHydCond, zScale_TOPMODEL
-  write(*,'(a,1x,5(f30.20,1x))') 'xDepth, zActive(ixSaturation)    = ', xDepth, zActive(ixSaturation)
-  write(*,'(a,1x,5(f30.20,1x))') 'xTran, trTotal(ixSaturation)     = ', xTran, trTotal(ixSaturation)
-  write(*,'(a,1x,5(f30.20,1x))') 'xFlow, totalColumnOutflow        = ', xFlow, sum(mLayerColumnOutflow(:))/HRUarea
-  !pause 'check groundwater'
- end if
```

Comment:
@martynpclark Can you also give an opinion on whether it's worth keeping (formal) debugging statements like these?

Reply:
The removal of this debugging statement was confirmed with @martynpclark in a meeting on June 6th because it is no longer used.